### DISCUSSION TITLE
📓 General

### DISCUSSION THREAD

**Discussion by ccali11** [consensusnetworks/casimir#203] [General]
##Enabling Casimir Front End for 12/6 Demo
**Get Balances**
- [ ] Should Hawyar query the blockchain (if he already has it indexed) instead of Chris iterating over each wallet provider bc Chris is limited by wallet provider api that only allows you to interface with one account at a time?

**Get amount Staked**
- [ ] Shane: are we showing how much they have staked to SSV only? Other pools/validators?
- [ ] Only ETH?
- [ ] Do we have the ability to show an address staking ETH to multiple pools? Specifically, do we have multiple abi's to deploy to?

**ETH Staking Popup Modal**
- [ ] Shane: where can we get the following info?
    - [ ] Date of Stake event
    - [ ] Distribution across pools <— Chris store this in local storage? Server? Shane other ideas?
    - [ ] Validator
    - [ ] Amount
    - [ ] Operator Score
    - [ ] Avg APR

**ETH Staking Fees**
- [ ] Chris get fees
- [ ] Expected Rewards —> Mock it?

**Comment by shanejearley** [comment]
> ## Enabling Casimir Front End for 12/6 Demo
> 
> **Get Balances**
> 
> Should Hawyar query the blockchain (if he already has it indexed) instead of Chris iterating over each wallet provider bc Chris is limited by wallet provider api that only allows you to interface with one account at a time?

Either 1) connect with one account at a time since we're demonstrating staking, or 2) mock user balance in existing local user implementation across multiple wallets using the 5 pre-populated addresses with 42 ETH. No, we won't use analytics service for this use at this point. Eventually yes, but start with either of the above. 

> 
> 
> 
> **Get amount Staked**
> 
> Shane: are we showing how much they have staked to SSV only? Other pools/validators?

Only ETH via SSV staking. 

> 
> Only ETH?

Yes. Stick to those 5 ETH wallets too for now (MM, CB, WC, Leg, Trez). 

> 
> Do we have the ability to show an address staking ETH to multiple pools? Specifically, do we have multiple abi's to deploy to?

One SSV manager contract to rule them all. Updating methods and responses in a PR shortly that will provide more info. 

> 
> 
> 
> **ETH Staking Popup Modal**
> 
> Shane: where can we get the following info?
> 
>     - Date of Stake event

Why do you need this? Showing a record log? Two events will be emitted by the manager contract 1) ManagerDeposit and 2) PoolStake - both happen once after a user deposits, and PoolStake will be emitted more than once if their stake is distributed to multiple pools in that deposit. Look into the ethers.Contract `.on`. These events emit block timestamps (or anything else we need). (Note, this data eventually gets indexed, cleaned and streamed through analytics, but if you need it, access it directly for now by listening for these events.)

> 
> Distribution across pools <— Chris store this in local storage? Server? Shane other ideas?

No server. Just use the contract method and events and store cleanly/locally. 

> 
>     - Validator
>     - Amount
>     - Operator Score
>     - Avg APR

Will provide the necessary info in a contract method response. You'll also need to look up each operator performance directly by the operator ID (returned in the pool response) using an SSV REST API. Will give more on this in the PR mentioned above.  

> 
> 
> 
> **ETH Staking Fees**
> 
> - Chris get fees
> 
> - Expected Rewards —> Mock it?

Yes, just mock to start so something like 4.56 or 5.43. 

> 
>

**Comment by ccali11** [comment]
### Front End Integration Notes

Add features:
- When user clicks on wallet provider and doesn't yet have it installed, direct user to external resource

Bugs:
- When clicking select wallet, bottom of page disappears
- When selecting a second address, previous provider's address remains

**Comment by DemogorGod** [comment]
Front End Notes

- fix line chart overflowing on screen size changes
- @shanejearley what is the minimum staking amount needed to integrate it with the front end since we are getting a minimum staking amount error 
- empty staking comp on staking action completion

**Reply by shanejearley** [reply]
0.0001 ETH (about 20 cents worth).
